### ISSUE TITLE
[feature] Add blank line to split title and description when possible

### ISSUE THREAD

**Issue by MatteoCampinoti94** [valentingol/docstripy#13]
As the title suggest, the rest output does not follow all PEP257 conventions and causes a D205 error when checked against it.

According to PEP257, There should be a blank line between the first text line and the rest of the docstring.

```python
def sort_list(l: list[int]) -> list[int]:
    """Return a sorted copy of the list.

    Sort the list in ascending order and return a copy of the
    result using the bubble sort algorithm.
    """
```

Ruff page: https://docs.astral.sh/ruff/rules/blank-line-after-summary

**Comment by valentingol** [comment]
Hello, thank you for your interest in the project! I know that a blank line is required to separate one-line title and short description. I don't add it until now as it can be tricky to add it in case of the title is in multiple lines. However, I agree that it could be a very nice feature to add, and I will try to make something that automatically detect when the title can be separate from the rest, taking account into the maximum length specified in config. Stay tuned!

**Comment by MatteoCampinoti94** [comment]
Thanks! I think ruff just checks if there is more than one full stop in the first line 🤔

So a very very basic solution could be this, but ideally it should be a regexp that splits around the first period that isn't inside quotes:

```python
title = ""
lines = [line.strip() for line in docstring.strip().split("\n")]
if lines:
    first_line, *lines = lines
    title, sep, second_line = first_line.partition(".")
    lines = [second_line, *lines] if sep else lines
```

**Comment by valentingol** [comment]
Yes I can try something like this and as you said take care of quotes and maybe periods without blank space right after, that cannot be interpreted as end of line like:
```python
class MyClass:
    def method(self) -> None:
        """Use MyClass.other_method to make something."""
```

**Comment by valentingol** [comment]
This is fixed in 0.7.1. There are many tricky situations where the split can fails, and I cover some of them. You can see some example in the [dedicated test fonction](https://github.com/valentingol/docstripy/blob/main/tests/unit_tests/test_title.py). Please re-open the issue if any problems

**Comment by MatteoCampinoti94** [comment]
Thank you!
